wpilibc/src/main/native/cpp/simulation/DriverStationSim.cpp

```diff
@@ -234,20 +234,20 @@ void DriverStationSim::SetJoystickType(int stick, int type) {
   HALSIM_SetJoystickType(stick, type);
 }
 
-void DriverStationSim::SetJoystickName(int stick, const char* name) {
-  HALSIM_SetJoystickName(stick, name);
+void DriverStationSim::SetJoystickName(int stick, std::string_view name) {
+  HALSIM_SetJoystickName(stick, name.data());
```

Comment:
We need to pass the size down to the HALSIM_ functions too and handle size appropriately there.